### ISSUE TITLE
Make it possible to execute tasks in parallel

### ISSUE THREAD

**Issue by mkollaro** [mkollaro/taskrunner#4]
This feature is currently not important, I'm not sure if I'll add it.

**Possible implementation:**
The execute function should be able to take a list of lists (of lists...) as a pipeline. Normally, they would just get all merged into a single list and executed in sequence, but there would be a special type derived from list (or just make it an object that contains a list), where the contents would get executed in parallel.

Example:
execute([taskA, parallel([taskB, taskC, taskD]), taskE)

In the previous case, taskE would get run only after all the previous ones.

Possible issues:
- how to make logging non-confusing?
- should the cleanups get run in parallel too?
- there are always a ton of problems with parallelism
